tests/by-util/test_tail.rs

```diff
@@ -4824,3 +4824,38 @@ fn test_obsolete_encoding_windows() {
         .stderr_is("tail: bad argument encoding: '-�b'\n")
         .code_is(1);
 }
+
+#[test]
+fn test_following_with_pid() {
+    use std::process::Command;
+
+    let scene = TestScenario::new(util_name!());
+
+    let sleep_command = Command::new("sleep")
+        .arg("999d")
+        .spawn()
+        .expect("failed to start sleep command");
+
+    let sleep_pid = sleep_command.id();
+
+    // when -f is specified, tail should die after
+    // the pid from --pid also dies
+    let mut child = scene
+        .ucmd()
+        .args(&["--pid", &sleep_pid.to_string(), "-f"])
+        .set_stdin(Stdio::null())
+        .stderr_to_stdout()
+        .run_no_wait();
+
+    child.make_assertion_with_delay(1000).is_alive();
+
+    Command::new("kill")
+        .arg("-9")
+        .arg(sleep_pid.to_string())
+        .output()
+        .expect("failed to kill sleep command");
+
+    child.make_assertion_with_delay(100).is_alive();
```

Comment:
0.1 seconds is a bit very short. Again, some CI runners are really slow, that's why I'm worried that this would result in a flaky test (we already have too many of them). Maybe increase it to 1 second, i.e. 1000 milliseconds?

Reply:
Alright. I'll go ahead and do that, then push. I'm usually of the mindset of trying to make sure any one test doesn't take too long, because it begins to add up over a large test suite. But I'd hate to make a weak or flaky test